test/unit/cdylib_name/cdylib_name_analysis_test.bzl

```diff
@@ -0,0 +1,42 @@
+"""Analysis tests for the name we assign to cdylib libraries."""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest", "asserts")
+load("//rust:defs.bzl", "rust_shared_library")
+
+def _cdylib_name_test_impl(ctx):
+    env = analysistest.begin(ctx)
+    target = analysistest.target_under_test(env)
+
+    file = target[DefaultInfo].files.to_list()[0]
```

Comment:
```suggestion
    # Expecting the `cdylib` to be the one and only file.
    files = target[DefaultInfo].files.to_list()
    asserts.equals(env, len(files), 1)
    file = files[0]
```

Reply:
on windows there will also be an interface library but I can do the appropriate check in each branch.

Reply:
I'm a big fan of checks that extract what's expected as explicitly as possible. If the files you're after aren't in an `OutputGroupinfo` provider or something, then I think there should be an assert that you can find a unique file from a depset. In general I've found asserting on the number of elements in the depset to cover most edge cases but can be annoying for volatile rules (don't think that's the case here).

Reply:
ok I've shuffled things a bit so that we check for the file count and all files in the group are what they are expected.